### PR TITLE
Functional Tests - Refacto search customer test

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Refacto search customer test
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| How to test?      | no need
| Possible impacts? | no

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27792)

### CODE REVIEW

File: tests/UI/pages/BO/orders/add.js [hunk 1]
```diff
@@ -131,6 +158,56 @@ class AddOrder extends BOBasePage {
...
+   * Get personal information title
+   * @param page {Page} Browser tab
+   * @param customerID {number} Id of customer to check details
+   * @returns {Promise<string>}
+   */
+  async getPersonalInformationTitle(page, customerID) {
+    const viewCustomerFrame = await page.frame({url: new RegExp(`sell/customers/${customerID}/view`, 'gmi')});
+
+    return viewCustomerPage.getPersonalInformationTitle(viewCustomerFrame);
+  }
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/orders/add.js]: Why not create a function to get the iframe, that call it directly on the test and use `viewCustomerPage` also on the test ?

Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/orders/add.js]: Thanks! good idea I will change it

File: tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+// Import test context
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_orders_orders_createOrders_searchViewCustomer';
+
+// Import expect from chai
+const {expect} = require('chai');
+
+let browserContext;
+let page;
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js:29]:
```suggestion
let page;
let customerIframe;
```

File: tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+      const isIframeVisible = await addOrderPage.clickOnDetailsButton(page);
+      await expect(isIframeVisible).to.be.true;
+    });
+
+    it('should check the existence of personal information block in the iframe', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkPersonalInformation', baseContext);
+
+      page = await addOrderPage.getCustomerIframe(page, DefaultCustomer.id);
+
+      const cardHeaderText = await viewCustomerPage.getPersonalInformationTitle(page);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js]:
```suggestion
      customerIframe = await addOrderPage.getCustomerIframe(page, DefaultCustomer.id);

      const cardHeaderText = await viewCustomerPage.getPersonalInformationTitle(customerIframe);
```

File: tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+      {args: {blockName: 'Vouchers', number: 0}},
+      {args: {blockName: 'Last emails', number: 0}},
+      {args: {blockName: 'Last connections', number: 0}},
+      {args: {blockName: 'Groups', number: 1}},
+      {args: {blockName: 'Addresses', number: 2}},
+    ].forEach((test) => {
+      it(`should check the ${test.args.blockName} number`, async function () {
+        await testContext.addContextItem(this, 'testIdentifier', `check${test.args.blockName}Number`, baseContext);
+
+        const cardHeaderText = await viewCustomerPage.getNumberOfElementFromTitle(page, test.args.blockName);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js]:
```suggestion
        const cardHeaderText = await viewCustomerPage.getNumberOfElementFromTitle(customerIframe, test.args.blockName);
```

File: tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+        await expect(parseInt(cardHeaderText, 10)).to.be.at.least(test.args.number);
+      });
+    });
+
+    it('should check the existence of add private note block', async function () {
+      await testContext.addContextItem(this, 'testIdentifier', 'checkAddPrivateNote', baseContext);
+
+      const isVisible = await viewCustomerPage.isPrivateNoteBlockVisible(page);
```
Comment (boubkerbribri) [inline tests/UI/campaigns/functional/BO/02_orders/01_orders/createOrders/01_searchViewCustomer.js]:
```suggestion
      const isVisible = await viewCustomerPage.isPrivateNoteBlockVisible(customerIframe);
```

**Review by boubkerbribri — COMMENTED** [review]
I suggest to create a var for the iframe, so it's more explicit to know that we are using it. And if want to add more steps outside the iframe, we have the page ready.

**Review by boubkerbribri — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @nesrineabdmouleh & @boubkerbribri

**Pull request merged**